### PR TITLE
Form Handlers improvements

### PR DESCRIPTION
**Pull request by simonsipl**
- changed default handler to text handler
- updated functional tests to match new text handler
- created form handlers page in documentation as it was lacking it

### CODE REVIEW

File: docs/form-handlers.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+##### `handleCheck(page, elementName, desiredValue)`
+- this method should contain a way of checking field e.g is field filled with provided text
+
+##### `getPriority()` 
+ - this method is used in automatioc field detection, higher value means 
+ isSatisifedBy method will be later
+
+
+## Example:
+```typescript
```
Comment (tgorskievo) [inline docs/form-handlers.md]: Add an example for Javascript too, as normally most users use JS in Kakunin projects.

Comment (simonsipl) [reply docs/form-handlers.md]: changed from TS example to JS example

File: docs/form-handlers.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
+---
+id: form-handlers
+title: Form Handlers
+---
+
+Form handlers allows you to fill or check if a element is filled with certain type of data.
```
Comment (tgorskievo) [inline docs/form-handlers.md]: 1. `allow` instead of `allows`
2. `an element` instead of `a element`

File: docs/form-handlers.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+For example you can fill select out of the box just by providing element and option.
+
+Using form handlers is very straightforward.
```
Comment (tgorskievo) [inline docs/form-handlers.md]: `the form handlers`

File: docs/form-handlers.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+```
+
+or with automatic field detection for example:
+
+```gherkin
+When I fill the "form" form with:
+            | descriptionTextarea | g:personalData:email |
+```
+
+also you can use form handlers to check if correct value is set:
```
Comment (tgorskievo) [inline docs/form-handlers.md]: `also, you can use form handlers to check if the correct value is set`

File: docs/form-handlers.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+```gherkin
+Then the "form" form is filled with:
+            | descriptionTextarea | v:storedTextareaValue |
+```
+
+
+Kakunin comes with a set of built in form handlers:
```
Comment (tgorskievo) [inline docs/form-handlers.md]: `built-in`

File: docs/form-handlers.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+## Custom Handlers
+
+We allows you to write your own handler, just by following
+form handlers interface
+
+## Methods:
+
+##### `isSatisfiedBy(element, elementName)`
+- this method is used to check if fill form step should use this strategy to for example to fill or select proper option
```
Comment (tgorskievo) [inline docs/form-handlers.md]: `select the proper `

File: docs/form-handlers.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+
+##### `handleFill(page, elementName, desiredValue)`
+- this method should contain a way of filling desired field e.g select proper option from custom Angular select
+
+##### `handleCheck(page, elementName, desiredValue)`
+- this method should contain a way of checking field e.g is field filled with provided text
```
Comment (tgorskievo) [inline docs/form-handlers.md]: `the provided text`

File: docs/form-handlers.md [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+
+`checkbox` - clicks desired checkbox option
+
+## Uploaded File Handler
+
+`uploadedFile` - this allows you to check if file is uploaded (this handler is only for checking)
+
+## Custom Handlers
+
+We allows you to write your own handler, just by following
```
Comment (tgorskievo) [inline docs/form-handlers.md]: `we allow`

File: docs/form-handlers.md [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+
+`CKEditor` - fills CKEditor field, with text
+
+## Radio Handler
+
+`radio` - clicks desired radio option 
+
+## Select Handler
+
+`select` - picks desired option from select
```
Comment (tgorskievo) [inline docs/form-handlers.md]: `the desired`

File: docs/form-handlers.md [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+
+Kakunin comes with a set of built in form handlers:
+
+## Text handler
+
+`text` - fills input
+
+## File Handler
+
+`file` - adds file, by providing file name (file must be in data folder)
```
Comment (tgorskievo) [inline docs/form-handlers.md]: `the data folder`

**Pull request merged**